package-res/resources/web/test/prompting/components/ScopedPentahoButtonComponentSpec.js

```diff
@@ -0,0 +1,71 @@
+/*!
+ * Copyright 2010 - 2015 Pentaho Corporation.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+define([ 'common-ui/prompting/components/ScopedPentahoButtonComponent', 'cdf/lib/jquery' ], function(
+  ScopedPentahoButtonComponent, $) {
+
+  describe("ScopedPentahoButtonComponent", function() {
+    it("should create component", function() {
+      var comp = new ScopedPentahoButtonComponent();
+      expect(comp.viewReportButtonRegistered).toBeFalsy();
+    });
+
+    describe("update", function() {
+      it("should register submit event", function() {
+        var comp = new ScopedPentahoButtonComponent();
+        spyOn(comp, "registerSubmitClickEvent");
+        comp.update();
+        expect(comp.registerSubmitClickEvent).toHaveBeenCalled();
+      });
+    });
+
+    describe("registerSubmitClickEvent", function() {
```

Comment:
This test feels too much like an integration test, since you are using jquery instead of spying on the funcitons inside of jquery which are being called. I would advise spying on those jquery functions and expecting that those be called instead of inspecting the DOM. At the very least, do not use jquery to verify the objects are present in the DOM, but instead use plain javascript.